database/sqlite/sqlite_functions.c

```diff
@@ -0,0 +1,1224 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "sqlite_functions.h"
+
+#define ENABLE_CACHE_CHARTS 1
+#define ENABLE_CACHE_DIMENSIONS 1
+
+const char *database_config[] = {
+    "PRAGMA auto_vacuum=incremental; PRAGMA synchronous=1 ; PRAGMA journal_mode=WAL; PRAGMA temp_store=MEMORY;",
+    "PRAGMA journal_size_limit=17179869184;",
```

Comment:
Well, 17GB of Journal size is a little bit much. How can we make sure this works in embedded devices?

Reply:
It's wrong -- I will switch to 16 or 32MB which is enough for the transactions we have at this stage for the WAL

Reply:
The worst case is probably going to be the huge transaction of parsing a metadata log file. Does this fit in 32MB?

Reply:
I will set it to 16MB. The largest size I observed by processing a single metadata logfile is about 2MB (each metadata logfile is  one transaction)